glusterd2/commands/volumes/volume-reset.go

```diff
@@ -0,0 +1,126 @@
+package volumecommands
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/glusterd2/peer"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/glusterd2/xlator"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+func registerVolResetStepFuncs() {
+	var sfs = []struct {
+		name string
+		sf   transaction.StepFunc
+	}{
+		{"vol-reset.Store", storeVolume},
+		{"vol-reset.NotifyVolfileChange", notifyVolfileChange},
+	}
+	for _, sf := range sfs {
+		transaction.RegisterStepFunc(sf.sf, sf.name)
+	}
+}
+
+func volumeResetHandler(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	volname := mux.Vars(r)["volname"]
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotFound)
+		return
+	}
+
+	var req api.VolOptionResetReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusUnprocessableEntity, errors.ErrJSONParsingFailed)
+		return
+	}
+
+	txn := transaction.NewTxn(ctx)
+	// Delete the option after checking for volopt flags
+	success := false
+	for _, k := range req.Options {
+		// Check if the key is set or not
+		if _, ok := volinfo.Options[k]; ok {
+			op, err := xlator.FindOption(k)
+			// If key exists, check for NEVER_RESET and FORCE flags
+			if err != nil {
+				logger.WithError(err)
+			} else if op.IsNeverReset() {
+				logger.WithError(err).Error("Option %s needs NEVER_RESET flag to be set", k)
```

Comment:
Http error required here. Return from here in case of failure. Message can be like "Reserved option, can't be reset"